graylog2-server/src/main/java/org/graylog2/rest/resources/system/SearchVersionResource.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.rest.resources.system;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.github.zafarkhaja.semver.expr.LexerException;
+import com.google.auto.value.AutoValue;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.graylog2.configuration.validators.SearchVersionRange;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.shared.rest.resources.RestResource;
+import org.graylog2.shared.utilities.StringUtils;
+import org.graylog2.storage.SearchVersion;
+import org.graylog2.storage.providers.ElasticsearchVersionProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.Locale;
+
+@Api(value = "System/SearchVersion", description = "Checks system search version requirements")
+@Path("/system/searchVersion")
+@Produces(MediaType.APPLICATION_JSON)
+@RequiresAuthentication
+public class SearchVersionResource extends RestResource implements PluginRestResource {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SearchVersionResource.class);
+    final private ElasticsearchVersionProvider versionProvider;
+
+    @Inject
+    public SearchVersionResource(ElasticsearchVersionProvider versionProvider) {
+        this.versionProvider = versionProvider;
+    }
+
+    @GET
+    @Path("/satisfiesVersion/{distribution}")
+    @ApiOperation(value = "Confirms whether the current search version satisfies a given distribution and Semantic Versioning version")
+    public SatisfiesVersionResponse satisfiesVersion(@ApiParam(name = "distribution", required = true) @PathParam("distribution") String distribution,
```

Comment:
Since `version` is optional it made it easier to allow it to be missing from the request by making it an optional query parameter instead of making multiple endpoints or trying to parse an optional extra path parameter using a RegEx. In the most recent commit, I have added a second endpoint that only takes a distribution and changes the queryparam here to a pathparam. Let me know what you think.

Reply:
Taking your explanation into account, I like your first version better ;-)

Reply:
I did too, but at least wanted to show one of the other options so there was something concrete to look at :). I will revert back to the other way shortly.